src/renderer/reader/index_reader.ts

```diff
@@ -78,6 +80,31 @@ ipcRenderer.on(readerIpc.CHANNEL,
                 }
                 data.payload.annotationTagsIndex = pushTags({}, annotationTagsList);
 
+
+                const bookmarkList = data.payload.reader.bookmark || [];
+                const bookmarkListLength = bookmarkList.length;
+
+                if (!data.payload.reader.bookmarkTotalCount) {
+                    data.payload.reader.bookmarkTotalCount = {
+                        state: bookmarkListLength,
+                    };
+                } else if (data.payload.reader.bookmarkTotalCount.state < bookmarkListLength) {
+                    data.payload.reader.bookmarkTotalCount.state = bookmarkListLength;
+                }
+                let bookmarkIndex = 0;
+                for (const [created, bookmark] of bookmarkList) {
+                    bookmarkIndex = bookmarkIndex + 1;
+                    if (!bookmark.created && created > bookmark.modified) {
+                        bookmark.created = bookmark.modified;
+                    }
+                    if (!bookmark.created) {
+                        bookmark.created = created;
+                    }
+                    if (!bookmark.name) {
+                        bookmark.name = `bookmark_${bookmarkIndex}`; // TODO? Probably translation issue !? 
```

Comment:
@danielweck do you have an idea to deal with bookmark no name migration ? In this PR we force the name of the bookmark to not rely of the bookmark index list position for the naming in UI if no name is available.

Reply:
I think the affordance that renders empty bookmark names (like empty annotation descriptions) must make it clear that this is empty. No need for a placeholder, let alone an arbitrary number bound to the list position (which can vary depending on filtering, ordering, etc.)

Reply:
seems fine, just this in this case : 
```
bookmark.name = `${bookmarkIndex}`;
```

Reply:
https://github.com/edrlab/thorium-reader/commit/6307e4ab4406b2af8ece81a3b19af64f173386ac